backend/TreeOfAKind.API/PersonsFiles/PersonsFilesController.cs

```diff
@@ -0,0 +1,135 @@
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TreeOfAKind.API.SeedWork;
+using TreeOfAKind.Application.Command;
+using TreeOfAKind.Application.Command.Trees.People.AddOrChangePersonsPhoto;
+using TreeOfAKind.Application.Command.Trees.People.AddPersonFile;
+using TreeOfAKind.Application.Command.Trees.People.RemovePersonsFile;
+using TreeOfAKind.Domain.Trees;
+using TreeOfAKind.Domain.Trees.People;
+
+namespace TreeOfAKind.API.PersonsFiles
+{
+    [Route("api/[controller]/[action]")]
+    public class PersonsFilesController : Controller
+    {
+        private readonly IMediator _mediator;
+
+        public PersonsFilesController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        /// <summary>
+        /// Adds file to person.
+        /// </summary>
+        /// <remarks>
+        /// Accepted content types are:
+        ///
+        ///     image/jpg
+        ///     image/jpeg
+        ///     image/png
+        ///     application/pdf
+        ///
+        /// Accepts body as from data!!!
+        /// </remarks>
+        /// <param name="request"></param>
+        /// <returns>Uri and Id of created file</returns>
+        /// <response code="201">File added to person</response>
+        /// <response code="400">Command is not valid</response>
+        /// <response code="401">User is not authenticated</response>
+        /// <response code="422">Business rule broken</response>
+        [HttpPost]
+        [Authorize]
+        [ProducesResponseType(typeof(IdUriDto),StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        public async Task<IActionResult> AddPersonsFile([FromForm] AddPersonsFileRequest request)
+        {
+            var authId = HttpContext.GetFirebaseUserAuthId();
+
+            var file = request.File;
+
+            var result = await _mediator.Send(new AddPersonsFileCommand(authId, new TreeId(request.TreeId),
+                new Document(file.OpenReadStream(), file.ContentType, file.Name), new PersonId(request.PersonId)));
+
+            return Created(string.Empty, new IdUriDto{Id = result.Id.Value, Uri = result.FileUri});
```

Comment:
TBH I don't know why would client need the file Uri right away

Reply:
Hmm idk, maybe, I'll leave it, u can ignore parameters when parsing :<